README.md

```diff
@@ -47,6 +47,76 @@ Click **Add user** and fill in the required details, and check the “**Allow AP
 
 For more instructions, check out examples folder.
 
+## A usage example for creating a server
+
+Below is an example configuration on how to create a server using the Terraform provider. Note that currently, only 10 concurrent servers creation is possible per account. For more examples, visit [official Terraform documentation](https://registry.terraform.io/providers/UpCloudLtd/upcloud/latest/docs).
+
+```
+provider "upcloud" {
+  # Your UpCloud credentials are read from the environment variables
+  # export UPCLOUD_USERNAME="Username for Upcloud API user"
+  # export UPCLOUD_PASSWORD="Password for Upcloud API user"
+}
+
+resource "upcloud_server" "server1" {
+  # System hostname
+  hostname = "terraform.example.com"
+
+  # Availability zone
+  zone = "nl-ams1"
+
+  # Number of CPUs and memory in GB
+  plan = "1xCPU-1GB"
+
+  storage_devices {
+    # System storage device size
+    size = 25
+
+    # Template UUID for Ubuntu 20.04
+    storage = "01000000-0000-4000-8000-000030200200"
+
+    # Storage device typeC
+    tier   = "maxiops"
+    action = "clone"
+  }
+
+  # Network interfaces
+  network_interface {
+    type = "public"
+  }
+
+  network_interface {
+    type = "utility"
+  }
+
+  # Include at least one public SSH key
+  login {
+    user = "root"
+    keys = [
+      "ssh-rsa public key",
+    ]
+    create_password = true
+    password_delivery = "email"
+  }
+
+  # Configuring connection details
+  connection {
+    # The server public IP address
+    host        = self.network_interface[0].ip_address
+    type        = "ssh"
+    user        = "root"
+    private_key = file("~/.ssh/rsa_private_key")
```

Comment:
This could be a bit confusing for users, since it fails the plan phase:
```
$ terraform plan

Error: Invalid function argument

  on test.tf line 63, in resource "upcloud_server" "server1":
  63:      private_key = file("~/.ssh/rsa_private_key")

Invalid value for "path" parameter: no file exists at
/Users/trdls/.ssh/rsa_private_key; this function works only with files that
are distributed as part of the configuration source code, so if this file will
be created by a resource in this configuration you must instead obtain this
result from an attribute of that resource.
```